### PR TITLE
Mmt-3117: Update MMT to use the new "Order Option Assignment" API

### PR DESCRIPTION
**Pull request by htranho**

### CODE REVIEW

**Review by jaybarra — APPROVED** [review]

File: app/controllers/manage_cmr/order_option_assignments_controller.rb [hunk 1]
```diff
@@ -92,7 +94,29 @@ def create
...
+  def destroy_order_option_assignments
+    success_count = 0
+    error_count = 0
+
+    params.fetch(:order_option_assignment, []).each do |assignment|
+      association = assignment.split('@')
+      collection_concept_id = association[0]
+      service_concept_id = association[1]
+
+      delete_service_assoc_response = cmr_client.delete_collection_service_association(service_concept_id: service_concept_id, collection_concept_id: collection_concept_id, token: token)
```
Comment (eudoroolivares2016) [inline app/controllers/manage_cmr/order_option_assignments_controller.rb:106]:
```suggestion
      # The order-option is on the association payload, deleting the association
      delete_service_assoc_response = cmr_client.delete_collection_service_association(service_concept_id: service_concept_id, collection_concept_id: collection_concept_id, token: token)
```

File: app/controllers/manage_cmr/order_option_assignments_controller.rb [hunk 2]
```diff
@@ -167,4 +303,44 @@ def get_order_options
...
+    order_option_list = Array.wrap(order_options_response.body.fetch('items', []))
+
+    order_option_select_values = []
+
+    order_option_list.select { |option| option.fetch('umm', {}).fetch('Deprecated', false) == false }.each do |order_option|
+      opt = [order_option.fetch('umm', {}).fetch('Name', ''), order_option.fetch('meta', {}).fetch('concept-id', '')]
+      order_option_select_values << opt
+    end
+
+    order_option_select_values.sort
```
Comment (eudoroolivares2016) [inline app/controllers/manage_cmr/order_option_assignments_controller.rb:323]: Is this going to sort by 'Name' or 'concept-id'

**Review by eudoroolivares2016 — APPROVED** [review]

File: app/controllers/manage_cmr/order_option_assignments_controller.rb (hunk 2 above)
Comment (htranho) [reply app/controllers/manage_cmr/order_option_assignments_controller.rb:323]: it sorts by name.

File: app/controllers/manage_cmr/order_option_assignments_controller.rb (hunk 1 above)
Comment (htranho) [reply app/controllers/manage_cmr/order_option_assignments_controller.rb:107]: Done, thanks

**Pull request merged**